src/components/CreateManifestForm/__tests__/CreateManifestForm.test.tsx

```diff
@@ -40,49 +45,249 @@ describe('Create Manifest Form', () => {
+          <CreateManifestForm {...props} />
+        </QueryClientProvider>
+      );
+      await new Promise((r) => setTimeout(r, 2000));
+      waitFor(() => expect(screen.findByText(`Manifest created`)).toBeInTheDocument());
+    });
   });
 
-  it('closes the create manifest modal on success', async () => {
-    const props = { ...createManifestFormProps, isSuccess: false };
+  describe('when the form has been successfully validated', () => {
+    it('displays helper text in name field ', async () => {
+      const props = { ...createManifestFormProps };
+      render(
+        <QueryClientProvider client={queryClient}>
+          <CreateManifestForm {...props} />
+        </QueryClientProvider>
+      );
+      const nameAlert = await screen.findAllByText(
+        'Your manifest name must be unique and must contain only numbers, letters, underscores, and hyphens.'
+      );
+      waitFor(() => expect(nameAlert).toHaveLength(1));
+    });
+  });
+  it('fires the submitForm when button is clicked', async () => {
+    const props = { ...createManifestFormProps };
 
-    const { rerender } = render(
+    const { getByText } = render(
       <QueryClientProvider client={queryClient}>
         <CreateManifestForm {...props} />
       </QueryClientProvider>
     );
 
-    expect(
-      screen.queryByText(
-        'Creating a new manifest allows you to export subscriptions to your on-premise subscription management application.'
-      )
-    ).toBeInTheDocument();
+    fireEvent.click(getByText('Create'));
+    waitFor(() => expect(submitForm).toHaveBeenCalledTimes(1));
+  });
+
+  describe('when both user form fields are validated', () => {
+    it('the create button should NOT be disabled', () => {
+      const props = { ...createManifestFormProps };
+
+      const { getByRole, getByText } = render(
+        <QueryClientProvider client={queryClient}>
+          <CreateManifestForm {...props} />
+        </QueryClientProvider>
+      );
+      const input = getByRole('textbox');
+      const createButton = getByText('Create');
+
+      expect(createButton).toBeInTheDocument();
+      expect(createButton).toHaveAttribute('disabled');
+
+      fireEvent.change(input, { target: { value: '123' } });
+
+      waitFor(() => {
+        expect(createButton).not.toHaveAttribute('disabled');
+      });
+    });
+  });
+
+  describe('name field on form', () => {
+    it('should handle change event', () => {
+      const props = { ...createManifestFormProps };
+      const handleNameChange = jest.fn();
+
+      act(() => {
+        render(
+          <QueryClientProvider client={queryClient}>
+            <CreateManifestForm {...props} />
+          </QueryClientProvider>
+        );
 
-    props.isSuccess = true;
+        const input = document.querySelector('textbox');
+        if (input) {
+          fireEvent.change(input, { target: { value: '' } });
+          expect(handleNameChange).toHaveBeenCalledTimes(1);
```

Comment:
I think we should check the result of this function being called, not that it has been